src/java.base/share/classes/jdk/internal/foreign/abi/BindingSpecializer.java

```diff
@@ -913,166 +891,54 @@ private static String valueLayoutConstantFor(Class<?> type) {
         }
     }
 
-    private static Class<?> valueLayoutTypeFor(Class<?> type) {
+    private static ClassDesc valueLayoutTypeFor(Class<?> type) {
         if (type == boolean.class) {
-            return ValueLayout.OfBoolean.class;
+            return CD_ValueLayout_OfBoolean;
         } else if (type == byte.class) {
-            return ValueLayout.OfByte.class;
+            return CD_ValueLayout_OfByte;
         } else if (type == short.class) {
-            return ValueLayout.OfShort.class;
+            return CD_ValueLayout_OfShort;
         } else if (type == char.class) {
-            return ValueLayout.OfChar.class;
+            return CD_ValueLayout_OfChar;
         } else if (type == int.class) {
-            return ValueLayout.OfInt.class;
+            return CD_ValueLayout_OfInt;
         } else if (type == long.class) {
-            return ValueLayout.OfLong.class;
+            return CD_ValueLayout_OfLong;
         } else if (type == float.class) {
-            return ValueLayout.OfFloat.class;
+            return CD_ValueLayout_OfFloat;
         } else if (type == double.class) {
-            return ValueLayout.OfDouble.class;
+            return CD_ValueLayout_OfDouble;
         } else if (type == MemorySegment.class) {
-            return AddressLayout.class;
+            return CD_AddressLayout;
         } else {
             throw new IllegalStateException("Unknown type: " + type);
         }
     }
 
-    private void emitInvokeStatic(Class<?> owner, String methodName, String descriptor) {
```

Comment:
It is great to see these ad-hoc routines (most of which exist in one form or another in all our JDK code generators) go away!